### ISSUE TITLE
Warnings as errors makes build fail (CMake)

### ISSUE THREAD

**Issue by robiwano** [graeme-hill/crossguid#30]
guid.cpp row 244:
```
// set all bytes to zero
void Guid::zeroify()
{
	std::fill(_bytes.begin(), _bytes.end(), 0);
}
```
should be:
```
// set all bytes to zero
void Guid::zeroify()
{
	std::fill(_bytes.begin(), _bytes.end(), static_cast<unsigned char>(0));
}
```
so that /WX option on windows doesn't make the build fail.

**Comment by robiwano** [comment]
This is on VS 2017 v15.5.7 btw.

**Comment by graeme-hill** [comment]
👍 Fixed in dec3a1b908c0a2aa273230ada96211a94da1a9b6
